code/datums/components/style/style.dm

```diff
@@ -0,0 +1,496 @@
+	RegisterSignal(parent, COMSIG_MOB_ITEM_ATTACK, PROC_REF(on_attack))
+	RegisterSignal(parent, COMSIG_HUMAN_MELEE_UNARMED_ATTACK, PROC_REF(on_punch))
+	RegisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH, PROC_REF(on_death))
+	RegisterSignal(parent, COMSIG_LIVING_RESONATOR_BURST, PROC_REF(on_resonator_burst))
+	RegisterSignal(parent, COMSIG_LIVING_PROJECTILE_PARRIED, PROC_REF(on_projectile_parry))
+	RegisterSignal(parent, COMSIG_LIVING_DEFUSED_GIBTONITE, PROC_REF(on_gibtonite_defuse))
+	RegisterSignal(parent, COMSIG_LIVING_CRUSHER_DETONATE, PROC_REF(on_crusher_detonate))
+	RegisterSignal(parent, COMSIG_LIVING_DISCOVERED_GEYSER, PROC_REF(on_geyser_discover))
+
+	projectile_parry = AddComponent(\
+		src,\
+		/datum/component/projectile_parry,\
+		list(\
+			/obj/projectile/colossus,\
+			/obj/projectile/temp/basilisk,\
+			/obj/projectile/kinetic,\
+			/obj/projectile/bileworm_acid,\
+			/obj/projectile/herald,\
+			)\
+		)
+
+
+
+/datum/component/style/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_ITEM_AFTERATTACK)
+	UnregisterSignal(parent, COMSIG_MOB_MINED)
+	UnregisterSignal(parent, COMSIG_MOB_APPLY_DAMAGE)
+	UnregisterSignal(parent, list(COMSIG_MOB_EMOTED("flip"), COMSIG_MOB_EMOTED("spin")))
+	UnregisterSignal(parent, list(COMSIG_MOB_ITEM_ATTACK, COMSIG_HUMAN_MELEE_UNARMED_ATTACK))
+	UnregisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH)
+	UnregisterSignal(parent, COMSIG_LIVING_RESONATOR_BURST)
+	UnregisterSignal(parent, COMSIG_LIVING_PROJECTILE_PARRIED)
+	UnregisterSignal(parent, COMSIG_LIVING_DEFUSED_GIBTONITE)
+	UnregisterSignal(parent, COMSIG_LIVING_DISCOVERED_GEYSER)
+
+	if(projectile_parry)
+		QDEL_NULL(projectile_parry)
+
+
+/datum/component/style/Destroy(force, silent)
+	STOP_PROCESSING(SSdcs, src)
+	var/mob/mob_parent = parent
+	if(mob_parent.hud_used)
+		mob_parent.hud_used.static_inventory -= meter
+		mob_parent.hud_used.show_hud(mob_parent.hud_used.hud_version)
+	return ..()
+
+
+/datum/component/style/process(seconds_per_tick)
+	point_multiplier = round(max(point_multiplier - 0.2 * seconds_per_tick, 1), 0.1)
+	change_points(-5 * seconds_per_tick * ROUND_UP((style_points + 1) / 200), use_multiplier = FALSE)
+	update_screen()
+
+
+
+/datum/component/style/proc/add_action(action, amount)
+	if(length(actions) > 9)
+		actions.Cut(1, 2)
+	if(length(actions))
+		var/last_action = actions[length(actions)]
+		if(action == actions[last_action])
+			amount *= 0.5
+	var/id
+	while(!id || (id in actions))
+		id = "action[rand(1, 1000)]"
+	actions[id] = action
+	change_points(amount)
+	addtimer(CALLBACK(src, PROC_REF(remove_action), id), 10 SECONDS)
+
+/datum/component/style/proc/remove_action(action_id)
+	actions -= action_id
+	update_screen()
+
+/datum/component/style/proc/change_points(amount, use_multiplier = TRUE)
+	if(!amount)
+		return
+
+	var/modified_amount = amount * (amount > 0 ? 1 - 0.1 * rank : 1) * (use_multiplier ? point_multiplier : 1)
+	style_points = clamp(style_points + modified_amount, -1, 499)
```

Comment:
you added a high score but its clamped to 499

Reply:
fixed